src/SecureHeaders.php

```diff
@@ -958,14 +954,19 @@ public function returnBuffer($buffer = null)
 
     private function importHeaders()
     {
-        # grab any headers that were already set and, if any, add these to our internal header list
+        $this->importStarted = true;
+        # grab any headers that were already set and, if any,
+        # add these to our internal header list
         $this->headerBag = $this->httpAdapter->getHeaders();
 
         foreach ($this->headerBag->get() as $header)
         {
             $this->addHeader($header->getName(), $header->getValue());
         }
 
+        # delete them (we'll set them again later)
+        $this->headerBag->removeAll();
```

Comment:
This avoids the headerBag receiving headers that it already has when headers are sent back to it (though the httpAdapter should still (as it currently does) remove headers from its header source before sending it new ones – for the same duplication reasons).